tribe-ext-ea-additional-options.php

```diff
@@ -0,0 +1,329 @@
+                    'tooltip' => esc_html__('Instead of linking to the Event page within The Events Calendar, enable this option so that visitors can be sent directly to the URL in the Website URL field. ** NOTE: This setting only affects legacy views and will not work in the upgraded views. **', 'tribe-ext-ea-additional-options'),
+                    'validation_type' => 'options',
+                    'default' => 'no',
+                    'options' => array(
+                        'no' => __('Link to the default single event page.', 'tribe-ext-ea-additional-options'),
+                        'yes' => __('Link directly to the event website URL', 'tribe-ext-ea-additional-options'),
+                    ),
+                ),
+                $this->opts_prefix . 'retain_line_breaks' => array(
+                    'type' => 'radio',
+                    'label' => esc_html__('Retain Line Breaks in Event Description', 'tribe-ext-ea-additional-options'),
+                    'tooltip' => esc_html__('Some import sources allow for linebreaks. Choose whether to remove linebreaks or keep them.', 'tribe-ext-ea-additional-options'),
+                    'validation_type' => 'options',
+                    'default' => 'no',
+                    'options' => array(
+                        'no' => __('Remove all line breaks from event descriptions.', 'tribe-ext-ea-additional-options'),
+                        'yes' => __('Retain all line breaks within event descirptions.', 'tribe-ext-ea-additional-options'),
+                    ),
+                ),
+            );
+
+            $setting_helper->add_fields(
+                    $fields, 'imports', // not the 'event-tickets' ("Tickets" tab) because it doesn't exist without Event Tickets
+                    'import-defaults-update_authority', true
+            );
+        }
+
+        /**
+         * Extension initialization and hooks.
+         */
+        public function init() {
+            // Load plugin textdomain
+            load_plugin_textdomain('tribe-ext-ea-additional-options', false, basename(dirname(__FILE__)) . '/languages/');
+
+            add_action('admin_init', array($this, 'add_settings'));
+
+            $deletionSetting = tribe_get_option($this->opts_prefix . 'delete_duplicate_removed_events');
+            if (!empty($deletionSetting) && $deletionSetting !== 'no') {
+                add_action('save_post_tribe-ea-record', array($this, 'record_finalized'), 10, 2);
+            }
+
+            add_filter('tribe_aggregator_url_import_range_options', array($this, 'add_other_url_options'));
+
+            add_action('tribe_events_aggregator_import_form_preview_options', array($this, 'add_import_options'));
+
+            add_filter('tribe_get_event_link', array($this, 'filter_event_link'), 100, 2);
+
+            add_filter('tribe_aggregator_event_translate_service_data_field_map', array($this, 'filter_service_data_field_map'));
```

Comment:
I see some conditionals inside of the callbacks ideally you can add the logic from the conditional here instead of the callback so in that case the callback is not executed when non required and you can remove the conditional from inside of the callback as well. For example.

```suggestion
            $lineBreakOpt = tribe_get_option($this->opts_prefix . 'retain_line_breaks');
            if ( tribe_is_truthy( $lineBreakOpt ) ) {
                   add_filter('tribe_aggregator_event_translate_service_data_field_map', array($this, 'filter_service_data_field_map'));
            }
```

Reply:
That also applies for other filters where you have conditionals inside of your functions.

Reply:
Not required but it would help to prevent execution of non required filters when the option was not enabled.

Reply:
Went through and applied this to all filters, when needed.